### PR TITLE
(feat) [SD-2250] Assign a default Source to user created content Items

### PR DESCRIPTION
**Pull request by sivakuna-aap**

### CODE REVIEW

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 1]
```diff
@@ -91,6 +91,11 @@
             this.stop(item);
             timeouts[item._id] = $timeout(function() {
                 var diff = extendItem({_id: item._id}, item);
+
+                if (diff.publish_schedule === null) {
```
Comment (amagdas) [inline client/app/scripts/superdesk-authoring/authoring.js]: Isn't this affecting scheduled publishing?

File: server/settings.py [hunk 2]
```diff
@@ -269,3 +269,6 @@ def env(variable, fallback_value=None):
 
 # Defines the maximum value of Publish Sequence Number after which the value will start from 1
 MAX_VALUE_OF_PUBLISH_SEQUENCE = 9999
+
+# Defines default value for Source to be set for manually created articles
+DEFAULT_SOURCE_VALUE_FOR_MANUAL_ARTICLES = 'AAP'
```
Comment (amagdas) [inline server/settings.py]: I think we should read this from system env so we can deploy for other envs without code changes.

**Comment by amagdas** [comment]
:+1:

File: client/app/scripts/superdesk-authoring/authoring.js (hunk 1 above)
Comment (sivakuna-aap) [inline client/app/scripts/superdesk-authoring/authoring.js]: No this won't affect. Without this autosave was throwing exceptions '`publish_schedule` is unknown field'. When I added in the schema it was throwing exception that null is not allowed in `publish_schedule`. That's the reason I had to add this check.

However, there is one more issue with published date for which I will raise JIRA.

Comment (amagdas) [inline client/app/scripts/superdesk-authoring/authoring.js]: :+1:

**Comment by akintolga** [comment]
:+1:

**Pull request merged**